### PR TITLE
Add vboxMargin, hboxMargin, vboxGrow, and hboxGrow helpers.

### PR DESCRIPTION
**Pull request by radicaled**
I'm actually not sure what the appropriate place for these helpers are -- I'm assuming its `Layouts.kt` since this does deal with layouts.

Somewhat mutilated example:

```kotlin
vbox {
    hboxGrow(Priority.ALWAYS)
    text("Header") {
        alignment = Pos.TOP_CENTER
        vboxMargin(bottom = 20.0)
    }
    listview(FXCollections.observableArrayList("Hello", "World"))
    buttonbar {
        vboxMargin(top = 10.0)
        button("Accept Entry")
        button("Reject Entry")
    }
}
```

### CODE REVIEW

**Comment by thomasnield** [comment]
Thanks @radicaled.  I think we implemented this already via hboxCobstraints and vboxConstraints.

**Comment by thomasnield** [comment]
See #12

**Comment by thomasnield** [comment]
And this chapter in the guide. Sorry for short messages.  On my phone. 

https://edvin.gitbooks.io/tornadofx-guide/content/Part%202%20-%20Usage%20Basics/7.%20Builders%20Part%20IV%20-%20Layouts%20and%20Menus.html

**Comment by radicaled** [comment]
Thanks, hadn't noticed. Not sure how I feel about the syntax, though.

```
hboxMargin(top = 5.0, bottom = 20.0)
```

feels a bit easier than

```
hboxConstraints {
  marginTop = 5.0
  marginBottom = 20.0
}
```

What about augmenting hboxConstraints?
```
hboxConstraints {
   setMargins(top = 5.0, bottom = 20.0)
}
```

Still a bit wordy, but less so. I guess I like the ability to define the margins all on one line, without having to go through the hassle of `Insets(0.0, 20.0, 0.0, 5.0)` and then trying to remember which positional argument corresponds to top/left/bottom/right.

(nice job on the book by the way, guys (and/or gals))

**Comment by thomasnield** [comment]
Thanks! I'm hoping we'll wrap it up completely by the end of the summer. 

Btw noticed a mistake in the guide. I'll correct my error later. It should be this:

```kotlin
vbox {
    button("Button 1") { 
        vboxConstraints {
            marginRight = 20.0
        vGrow = Priority.ALWAYS
    }
    }
    button("Button 2")
}
```

Regarding the augmentation, I don't have too strong of a preference. @t-boom and @edvin what do you think?

**Comment by t-boom** [comment]
I also don't really care much either way. I don't feel the current method is that bad.

**Comment by edvin** [comment]
Sure it could be improved slightly but I don't think we should break backwards compatibility because of it. I also don't like multiple ways of expressing the same thing if we can avoid it.

**Comment by edvin** [comment]
I also tend to use inline type safe styles instead of setting layout properties where applicable, so I don't really use these functions a lot :)

**Comment by radicaled** [comment]
Fair enough!

Thanks for the feedback, guys!
